recipes/mosquitto/2.x/conanfile.py

```diff
@@ -55,82 +76,101 @@ def _configure_cmake(self):
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "res"), "*.example")
         if not self.options.shared:
             tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.so*")
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dll*")
-            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dylib*")
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.dylib")
+            tools.remove_files_by_mask(os.path.join(self.package_folder, "bin"), "*.dll")
+        elif self.options.shared and self.settings.compiler == "Visual Studio":
+            self.copy("mosquitto.lib", src=os.path.join(self._build_subfolder, "lib"), dst="lib")
+            if self.options.build_cpp:
+                self.copy("mosquittopp.lib", src=os.path.join(self._build_subfolder, "lib"), dst="lib")
 
     def package_info(self):
-        lib_suffix = "_static" if not self.options.shared else ""
-        # FIXME: there should be no namespace for CMake targets
-        self.cpp_info.components["libmosquitto"].name = "mosquitto"
-        self.cpp_info.components["libmosquitto"].libs = ["mosquitto" + lib_suffix]
-        self.cpp_info.components["libmosquitto"].names["pkgconfig"] = "libmosquitto"
-        if self.options.with_tls:
-            self.cpp_info.components["libmosquitto"].requires.append("openssl::openssl")
-            self.cpp_info.components["libmosquitto"].defines.append("WITH_TLS")
-        if self.settings.os == "Windows":
-            self.cpp_info.components["libmosquitto"].system_libs.append("ws2_32")
-            if self.options.with_tls:
-                self.cpp_info.components["libmosquitto"].system_libs.append("crypt32")
-        elif self.settings.os == "Linux":
-            self.cpp_info.components["libmosquitto"].system_libs.extend(["rt", "pthread", "dl"])
-
-        self.cpp_info.components["libmosquittopp"].name = "mosquittopp"
-        self.cpp_info.components["libmosquittopp"].libs = ["mosquittopp" + lib_suffix]
-        self.cpp_info.components["libmosquittopp"].requires = ["libmosquitto"]
-        self.cpp_info.components["libmosquittopp"].names["pkgconfig"] = "libmosquittopp"
-        if not self.options.shared:
-            self.cpp_info.components["libmosquitto"].defines.append("LIBMOSQUITTO_STATIC")
-
-        bin_path = os.path.join(self.package_folder, "bin")
-        self.output.info("Appending PATH env var with : {}".format(bin_path))
-        self.env_info.PATH.append(bin_path)
+        libsuffix = "" if self.options.shared else "_static"
+        self.cpp_info.components["libmosquitto"].names["pkg_config"] = "libmosquitto"
+        self.cpp_info.components["libmosquitto"].libs = ["mosquitto" + libsuffix]
+        self.cpp_info.components["libmosquitto"].requires = ["openssl::openssl"]
```

Comment:
Should this be conditional?

```suggestion
        if self.options.ssl:
            self.cpp_info.components["libmosquitto"].requires = ["openssl::openssl"]
```